src/utils/keyboardKey.js

```diff
@@ -0,0 +1,295 @@
+ * Unfortunately, `key` is not yet fully supported.
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key
+ */
+import _ from 'lodash'
+
+export const codes = {
+  // ----------------------------------------
+  // By Code
+  // ----------------------------------------
+  3: 'Cancel',
+  6: 'Help',
+  8: 'Backspace',
+  9: 'Tab',
+  12: 'Clear',
+  13: 'Enter',
+  16: 'Shift',
+  17: 'Control',
+  18: 'Alt',
+  19: 'Pause',
+  20: 'CapsLock',
+  27: 'Escape',
+  28: 'Convert',
+  29: 'NonConvert',
+  30: 'Accept',
+  31: 'ModeChange',
+  32: ' ',
+  33: 'PageUp',
+  34: 'PageDown',
+  35: 'End',
+  36: 'Home',
+  37: 'ArrowLeft',
+  38: 'ArrowUp',
+  39: 'ArrowRight',
+  40: 'ArrowDown',
+  41: 'Select',
+  42: 'Print',
+  43: 'Execute',
+  44: 'PrintScreen',
+  45: 'Insert',
+  46: 'Delete',
+  48: ['0', ')'],
+  49: ['1', '!'],
+  50: ['2', '@'],
+  51: ['3', '#'],
+  52: ['4', '$'],
+  53: ['5', '%'],
+  54: ['6', '^'],
+  55: ['7', '&'],
+  56: ['8', '*'],
+  57: ['9', '('],
+  91: 'OS',
+  93: 'ContextMenu',
+  144: 'NumLock',
+  145: 'ScrollLock',
+  181: 'VolumeMute',
+  182: 'VolumeDown',
+  183: 'VolumeUp',
+  186: [';', ':'],
+  187: ['=', '+'],
+  188: [',', '<'],
+  189: ['-', '_'],
+  190: ['.', '>'],
+  191: ['/', '?'],
+  192: ['`', '~'],
+  219: ['[', '{'],
+  220: ['\\', '|'],
+  221: [']', '}'],
+  222: ["'", '"'],
+  224: 'Meta',
+  225: 'AltGraph',
+  246: 'Attn',
+  247: 'CrSel',
+  248: 'ExSel',
+  249: 'EraseEof',
+  250: 'Play',
+  251: 'ZoomOut',
+}
+
+// Function Keys (F1-24)
+_.times(24, (i) => codes[112 + i] = `F${i + 1}`)
+
+// Alphabet (a-Z)
+_.times(26, (i) => {
+  const n = i + 65
+  codes[n] = [String.fromCharCode(n + 32), String.fromCharCode(n)]
+})
+
+const keyboardKey = {
+  /**
+   * Get the `keyCode` or `which` value from a keyboard event or `key` name.
+   * @param {string|object} name A keyboard event like object or `key` name.
+   * @param {string} [name.key] If object, it must have one of these keys.
+   * @param {string} [name.keyCode] If object, it must have one of these keys.
+   * @param {string} [name.which] If object, it must have one of these keys.
+   * @returns {*}
+   */
+  getCode(name) {
+    return _.isObject(name) ? name.keyCode || name.which || this[name.key] : this[name]
+  },
+
+  /**
+   * Get the key name from a keyboard event, `keyCode`, or `which` value.
+   * @param {number|object} code A keyboard event like object or key name.
+   * @param {number} [code.keyCode] If object, it must have one of these keys.
+   * @param {number} [code.which] If object, it must have one of these keys.
+   * @param {number} [code.shiftKey] If object, it must have one of these keys.
+   * @returns {*}
+   */
+  getName(code) {
+    const isEvent = _.isObject(code)
+    let name = codes[isEvent ? code.keyCode || code.which : code]
+
+    if (Array.isArray(name)) {
+      if (isEvent) {
+        name = name[+code.shiftKey]
```

Comment:
Oh boy, never seen that done before, if that says anything about readability. Your call.